physics/src/modules/include/ILateralIceSpread.hpp

```diff
@@ -20,53 +20,54 @@ class ILateralIceSpread : public ModelComponent {
     virtual ~ILateralIceSpread() = default;
 
     std::string getName() const override { return "LateralIceSpread"; }
-    void setData(const ModelState::DataMap& ms) override { }
-    /*!
-     * Updates the freezing of open water for the timestep.
-     *
-     * @param tStep The object containing the timestep start and duration times.
-     * @param hice The ice-average ice thickness.
-     * @param hsnow The ice-average snow thickness.
-     * @param deltaHi The change in ice thickness this timestep.
-     * @param newIce The positive change in ice thickness this timestep.
-     * @param cice The ice concentration.
-     * @param qow The open water heat flux.
-     * @param deltaCFreeze The change in concentration due to freezing.
-     */
-    virtual void freeze(const TimestepTime& tstep, double hice, double hsnow, double deltaHi,
-        double newIce, double& cice, double& qow, double& deltaCfreeze)
-        = 0;
+    void setData(const ModelState::DataMap& ms) override
+    {
+        deltaCIce.resize();
+        newice.resize();
+        snowMelt.resize();
+
+        /* Set these to zero, so we don't have uninitialized values floating around.
+         * Should this be done here, or in derived classes? Arguably only in the derived classes,
+         * since having uninitialised values helps with error detection. But as long as we have the
+         * nextsim_thermo.use_thermo_forcing option, all derived classes must initialise these
+         * values to something sensible.*/
```

Comment:
I'd limit this comment to what the code is doing and why. And leave out the uncertainty.
```
         /*
           * Set these to zero, so we don't have uninitialized values floating around.
           * As long as we have the nextsim_thermo.use_thermo_forcing option, all
           * derived classes must initialise these values to something sensible.
           */